common/djangoapps/student/management/commands/export_staff_users.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import absolute_import, print_function
+
+import logging
+from datetime import datetime, timedelta
+from django.core.management.base import BaseCommand
+from django.conf import settings
+from django.core.mail.message import EmailMultiAlternatives
+from django.template.loader import get_template
+from os import remove
+
+from djqscsv import write_csv
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from student.models import CourseAccessRole
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Example usage:
+        $ ./manage.py lms export_staff_users -d 7 --settings=devstack_docker
+        $ ./manage.py lms export_staff_users --days 7 --settings=devstack_docker
+    """
+
+    help = """
+    This command will export a csv of all users who have logged in within the given days and
+    have staff access role in active courses (Courses with end date in the future).
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '-d',
+            '--days',
+            type=int,
+            default=7,
+            help='Indicate the login time period in days starting from today'
+        )
+
+    subject = 'Staff users CSV'
+    to_addresses = ['aazam@edx.org']
+    from_address = settings.DEFAULT_FROM_EMAIL
+    txt_template_path = 'email/export_staff_users.txt'
+    html_template_path = 'email/export_staff_users.html'
+    csv_filename = 'staff_users.csv'
+
+    def handle(self, *args, **kwargs):
+        days = kwargs['days']
+        current_date = datetime.now()
+        starting_date = current_date - timedelta(days=days)
+        active_courses = CourseOverview.objects.filter(end__gte=current_date).values_list('id', flat=True)
+        course_access_roles = CourseAccessRole.objects.filter(
+            role__in=['staff', 'instructor'],
+            user__last_login__range=(current_date, starting_date),
+            course_id__in=active_courses,
+            user__is_staff=False
+        ).values('user__username', 'user__email', 'role')
+        try:
+            self.send_email(course_access_roles, days)
+            logger.info(
+                'Sent staff users email for the period {} to {}. Staff users count:{}'.format(
+                    starting_date,
+                    current_date,
+                    course_access_roles.count()
+                )
+            )
+        except Exception:
+            logger.exception(
+                'Failed to send staff users email for the period {}-{}'.format(starting_date, current_date)
+            )
+
+        return course_access_roles.count()
```

Comment:
What is the purpose of returning `course_access_roles.count()`?

Reply:
I am using this for assertion in the unit test